.github/workflows/pre-merge.yaml

```diff
@@ -29,6 +29,12 @@ jobs:
           restore-keys: |
             cache-gradle-
 
+      - name: Setup Java Version
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '11'
```

Comment:
shouldn't be behind a `${{ matrix.java }}` too? when it runs `./gradlew assembleRelease` using Java 11 it might fail

Reply:
> using Java 11 it might fail

Why this?

Reply:
just guessing here, Android always required Java 8 or am I missing something?

Reply:
and if it's not the case, why do we need a Java matrix in the `test-publish-dry-run.yaml` file then? Java 11 would work for every use case

Reply:
`test-publish-dry-run.yaml` is more testing from the users prospective. We want to try on an environment that is similar to the potential user machine, therefore having Java 8 on pre-AGP7 is reasonable.

The pre-merge workflow is instead building the project + running the Gradle runner tests. We necessarily need Java 11 here as on of those test is doing a run on AGP7 and will fail if run on Java8.

Reply:
makes sense, and CI is green so all good